src/resolvers/Mutation/updateAdvertisement.ts

```diff
@@ -0,0 +1,125 @@
+import type { MutationResolvers } from "../../types/generatedGraphQLTypes";
+import { Advertisement, User } from "../../models";
+import { errors, requestContext } from "../../libraries";
+import {
+  ADVERTISEMENT_NOT_FOUND_ERROR,
+  USER_NOT_FOUND_ERROR,
+  INPUT_NOT_FOUND_ERROR,
+  END_DATE_VALIDATION_ERROR,
+  START_DATE_VALIDATION_ERROR,
+  FORBIDDEN_FIELD_UPDATE_ERROR,
+  FIELD_NON_EMPTY_ERROR,
+} from "../../constants";
+
+export const updateAdvertisement: MutationResolvers["updateAdvertisement"] =
+  async (_parent, args, _context) => {
+    const { _id, ...otherFields } = args.input;
+
+    //If there is no input
+    if (Object.keys(otherFields).length === 0) {
+      throw new errors.InputValidationError(
+        requestContext.translate(INPUT_NOT_FOUND_ERROR.MESSAGE),
+        INPUT_NOT_FOUND_ERROR.CODE,
+        INPUT_NOT_FOUND_ERROR.PARAM
+      );
+    }
+
+    const permittedFields = [
+      "_id",
+      "name",
+      "link",
+      "type",
+      "startDate",
+      "endDate",
+    ];
+
+    // Check for unintended null values in permitted fields, if all fields are permitted
+    for (const field of Object.keys(args.input)) {
+      if (!permittedFields.includes(field)) {
```

Comment:
this check is useless, graphql only accepts fields that are defined in the schema